docs/topics/scripting/custom-script-deps-tutorial.md

```diff
@@ -0,0 +1,332 @@
+
+   ```kotlin
+    fun main(vararg args: String) {
+        if (args.size != 1) {
+            println("usage: <app> <script file>")
+        } else {
+            val scriptFile = File(args[0])
+            println("Executing script $scriptFile")
+            val res = evalFile(scriptFile)
+            res.reports.forEach {
+                if (it.severity > ScriptDiagnostic.Severity.DEBUG) {
+                    println(" : ${it.message}" + if (it.exception == null) "" else ": ${it.exception}")
+                }
+            }
+        }
+    }
+   ```
+
+You can find the full code [here](https://github.com/Kotlin/kotlin-script-examples/blob/master/jvm/basic/jvm-maven-deps/host/src/main/kotlin/org/jetbrains/kotlin/script/examples/jvm/resolve/maven/host/host.kt)
+
+## Run scripts
+
+To check how your scripting host works, prepare a script to execute and a run configuration.
+
+1. Create the file `html.scriptwithdeps.kts` with the following content in the project root directory:
+
+   ```kotlin
+   @file:Repository("https://maven.pkg.jetbrains.space/public/p/kotlinx-html/maven")
+   @file:DependsOn("org.jetbrains.kotlinx:kotlinx-html-jvm:0.7.3")
+   
+   import kotlinx.html.*; import kotlinx.html.stream.*; import kotlinx.html.attributes.*
+   
+   val addressee = "World"
+   
+   print(
+       createHTML().html {
+           body {
+               h1 { +"Hello, $addressee!" }
+           }
+       }
+   )
+   ```
+   
+   It uses functions from the `kotlinx-html-jvm` library which is referenced in the `@DependsOn` annotation argument.
+
+2. Create a run configuration that starts the scripting host and executes this file:
+   1. Open `host.kt` and navigate to the `main` function. It has a **Run** gutter icon on the left.
+   2. Right-click the gutter icon and select **Modify Run Configuration**.
+   3. In the **Create Run Configuration** dialog, add the script file name to **Program arguments** and click **OK**.
+   
+      ![Scripting host run configuration](script-deps-run-config.png){width=800}
+
+3. Run the created configuration.
+
+You'll see how the script is executed: it resolves the dependency on `kotlinx-html-jvm` in the specified repository and
+prints the results of calling its functions:
+
+```text
+<html>
+  <body>
+    <h1>Hello, World!</h1>
+  </body>
+</html>
+```
+
+Resolving dependencies may take some time on the first run. Subsequent runs will complete much faster because they use
+downloaded dependencies from the local Maven repository.
+
+## What's next?
+
+Once you've created a simple Kotlin scripting project, find more information on this topic:
+* Read the [Kotlin scripting KEEP](https://github.com/Kotlin/KEEP/blob/master/proposals/scripting-support.md)
+* Browse more [Kotlin scripting examples](https://github.com/Kotlin/kotlin-script-examples)
+* Watch the talk [Implementing the Gradle Kotlin DSL](https://kotlinconf.com/2019/talks/video/2019/126701/) by Rodrigo Oliveira
```

Comment:
Also we shoul think about adding a link to this article to other docs places (like this https://kotlinlang.org/docs/command-line.html#run-scripts)

Reply:
Added